pyne/vta.py

```diff
@@ -0,0 +1,500 @@
+PyNE is still in need.
+
+.. moduleauthor:: X. Zhang
+"""
+from __future__ import division
+import math
+import numpy as np
+
+
+##################################
+### Facet voxel traverse tools ###
+##################################
+
+
+def _is_point_in_mesh(bounds, point):
+    """
+    Check whether a point is in the mesh.
+
+    Parameters:
+    -----------
+    bounds : list
+        Boundaries. Nested list of the boundaries along x, y, and z directions.
+        Expected format: [[x_min, ..., x_max],
+                          [y_min, ..., y_max],
+                          [z_min, ..., z_max]]
+    point : numpy array
+        An array with size of 3, represents the point to be checked.
+
+    Return:
+    -------
+    True : when point in the mesh.
+    False : when point outside the mesh.
+    """
+    if bounds[0][0] <= point[0] <= bounds[0][-1] and \
+       bounds[1][0] <= point[1] <= bounds[1][-1] and \
+       bounds[2][0] <= point[2] <= bounds[2][-1]:
+        return True
+    else:
+        return False
+
+def _find_voxel_idx_1d(bounds_1d, cor, vec_1d):
+    """
+    Find the voxel id (in a given dimension).
+
+    Parameters:
+    -----------
+    bounds_1d : list
+        Boundary of the mesh in a given direction.
+    cor : float
+        Coordinate of the point in the given direction.
+    vec_1d : float
+        Vector attribute in the given direction. Used for determining the
+        search direction.
+
+    Return:
+    -------
+    idx : int
+        Mesh index of the given direction. This value is -1 when the point not
+        in the mesh.
+    ----------
+    """
+    idx = -1
+    bounds_1d = np.array(bounds_1d)
+    if vec_1d > 0:
+        idx = np.searchsorted(bounds_1d, cor, side='right')
+    else:
+        idx = np.searchsorted(bounds_1d, cor, side='left')
+    if idx == 0 or idx == len(bounds_1d):
+    # idx could be -1, that means point outside the mesh
+        return -1
+    else:
+        return idx - 1
+
+def _calc_vec_dir(start, end):
+    """
+    Calculate the direction from start to end (start -> end).
+    
+    Parameters:
+    -----------
+    start : numpy array
+        An array of size 3. The start point.
+    end :
+        An array of size 3. The end point.
+
+    Return:
+    -------
+    vec : numpy array
+        Direction from start to end, an unit vector.
+    """
+    return np.divide(end-start, np.linalg.norm(end-start))
+
+def _find_next_grid_1d(cor, vec_1d, bounds_1d):
+    """
+    Find the next grid coordinate of a direction.
+
+    Parameters:
+    -----------
+    cor : float
+        Coordinate of current point in the direction.
+    vec_1d : float
+        Direction value.
+    bounds_1d : list of float
+        Boundary values of current direction.
+
+    Return:
+    -------
+    n_grid : float
+        The coordinate value of next mesh grid.
+    """
+    voxel_idx = _find_voxel_idx_1d(bounds_1d, cor, vec_1d)
+    if voxel_idx == -1:
+        return vec_1d * float('inf')
```

Comment:
Why not np.inf?. [https://stackoverflow.com/questions/42315541/difference-between-np-inf-and-floatinf](https://stackoverflow.com/questions/42315541/difference-between-np-inf-and-floatinf)